src/System.Private.ServiceModel/src/System/ServiceModel/Channels/TransportSecurityBindingElement.cs

```diff
@@ -49,10 +51,113 @@ internal override bool SupportsRequestReply
             get { return true; }
         }
 
+        internal override SecurityProtocolFactory CreateSecurityProtocolFactory<TChannel>(BindingContext context, SecurityCredentialsManager credentialsManager, bool isForService, BindingContext issuerBindingContext)
+        {
+            if (context == null)
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("context");
+            if (credentialsManager == null)
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("credentialsManager");
```

Comment:
Possibly -- but this is copied from the full framework.  I traced all the calls it makes and I believe none assert all the way down till the point is really is dereferenced.  But I'd rather not start cleaning up the code copied from the full framework at the risk of making this PR even larger.

---
In reply to: [78346772](https://github.com/dotnet/wcf/pull/1515#discussion_r78346772) [](ancestors = 78346772)